airbyte-integrations/connectors/source-datadog/source_datadog/source.py

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from datetime import datetime
+from json.decoder import JSONDecodeError
+from typing import Any, List, Mapping, Optional, Tuple
+
+import requests
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from pydantic.datetime_parse import timedelta
+from source_datadog.streams import AuditLogs, Dashboards, Downtimes, Incidents, IncidentTeams, Logs, Metrics, SyntheticTests, Users
+
+
+class SourceDatadog(AbstractSource):
+    @staticmethod
+    def _get_authenticator(config: Mapping[str, Any]):
+        return DatadogAuthenticator(api_key=config["api_key"], application_key=config["application_key"])
+
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
+        alive = True
+        error_msg = None
+
+        try:
+            args = self.connector_config(config)
+            dd_dashboards = Dashboards(**args)
+            for _ in dd_dashboards.read_records(sync_mode=SyncMode.full_refresh):
+                continue
+        except ConnectionError as error:
+            alive, error_msg = False, repr(error)
+        except JSONDecodeError:
+            alive, error_msg = (
+                False,
+                "Unable to connect to the Datadog API with the provided credentials. Please make sure the input "
+                "credentials and environment are correct.",
+            )
+
+        return alive, error_msg
```

Comment:
can you check how other sources do the `check_connection` you can implement in a simpler and cleaner way this.

Reply:
Updated @marcosmarxm